Dockerfile

```diff
@@ -1,29 +1,40 @@
+# syntax=docker/dockerfile:1.4
 # This Dockerfile requires BuildKit to be enabled, by setting the environment variable
 # DOCKER_BUILDKIT=1
 # see https://docs.docker.com/build/buildkit/#getting-started
 ARG BASE_DIGEST_AMD64="sha256:d1c582c712e8ae049014a3e81296c522360949cbac7ade7e50a0d4d310706066"
 ARG BASE_DIGEST_ARM64="sha256:dac99a547dae7b46de7f0c34f57002094e054724058e8c9f51243d53286bf89e"
 
-#### Builder image ####
-FROM ubuntu:jammy as builder
-ARG DISTBALL
+# set to "build" to build zeebe from scratch instead of using a distball
+ARG DIST="distball"
 
-ENV TMP_ARCHIVE=/tmp/zeebe.tar.gz \
-    TMP_DIR=/tmp/zeebe
-
-COPY ${DISTBALL} ${TMP_ARCHIVE}
-
-RUN mkdir -p ${TMP_DIR} && \
-    tar xfvz ${TMP_ARCHIVE} --strip 1 -C ${TMP_DIR} && \
-    # already create volume dir to later have correct rights
-    mkdir ${TMP_DIR}/data && \
+### Init image containing tini and the startup script ###
+FROM ubuntu:jammy as init
+WORKDIR /
+RUN --mount=type=cache,target=/var/apt/cache,rw \
     apt-get -qq update && \
     apt-get install -y --no-install-recommends tini=0.19.0-1 && \
-    cp /usr/bin/tini ${TMP_DIR}/bin/tini
+    cp /usr/bin/tini .
+COPY --link --chown=1000:0 docker/utils/startup.sh .
+
+### Build zeebe from scratch ###
+FROM maven:3-eclipse-temurin-17 as build
+WORKDIR /
```

Comment:
🔧  was wondering whether we should rather have dedicated workdir (e.g. `/zeebe`), than can be assumed to be "clean", this way everything is dumped into the root of this build container?